.github/workflows/test.yml

```diff
@@ -0,0 +1,105 @@
+name: Test
+
+on:
+  pull_request:
+  push:
+    branches:
+    - develop
+    - main
+    tags: '[0-9]+.[0-9]+.[0-9]+'
+    paths-ignore:
+    - 'README.md'
+    - 'CODE_OF_CONDUCT.md'
+    - 'CONTRIBUTING.md'
+    - 'LICENSE'
+    - 'SECURITY.md'
+    - 'docs/**'
+    - '.github/**'
+    - '!.github/workflows/test.yml'
+  release:
+    types:
+    - created
+
+# Keep in sync with codeql-analysis.yml and build.yml
+env:
+  CI: true
+  node: 14.x
+  java: 15
+
+jobs:
+
+  server-tests:
+    if: github.event_name != 'release'
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Java
+      uses: actions/setup-java@v1
+      with:
+        java-version: '${{ env.java }}'
+    - uses: actions/checkout@v2
+    - name: Cache Gradle dependencies
+      uses: actions/cache@v2
+      with:
+        path: ~/.gradle/caches
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+        restore-keys: |
+          ${{ runner.os }}-gradle-
+    - name: Java Tests
+      run: ./gradlew --console=plain executeTests jacocoTestReport -x yarn -x webpack jacocoTestCoverageVerification
+    - name: "Codacy: Report coverage"
+      uses: codacy/codacy-coverage-reporter-action@master
+      with:
+        project-token: ${{ secrets.CODACY_PROJECT_TOKEN }}
+        coverage-reports: build/reports/jacoco/test/jacocoTestReport.xml
+      if: (github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name) && (success() || failure())
+    - name: Java Code Style
+      run: ./gradlew spotlessCheck
+      if: success() || failure()
+    - name: Java Documentation
+      run: ./gradlew checkstyleMain -x yarn -x webpack
+      if: success() || failure()
```

Comment:
Can we move Spotless and Checkstyle in front of the testing step? I think we should fail fast here in case something is wrong because the tests take a lot more time to run through.

Reply:
Nice Idea. I separated the linting and tests in different jobs. :+1: